ambry-rest/src/main/java/com/github/ambry/rest/PublicAccessLogHandler.java

```diff
@@ -107,6 +108,10 @@ public void write(ChannelHandlerContext ctx, Object msg, ChannelPromise promise)
         logger.error(
             "Sending response that is not of type HttpResponse or HttpContent. Sending response to {}. Request is of type {}. No action being taken other than logging this unexpected state.",
             ctx.channel().remoteAddress(), msg.getClass());
+      } else {
```

Comment:
this may not catch the (rarer) case of use `FullHttpResponse` which implements both `HttpResponse` and `HttpContent`.

What if we changed it to this:
```
boolean isHttpContent = msg instanceof HttpContent;
if (msg instanceof HttpResponse) {
  ...
} else if (!isHttpContent) {
 log
}
if (isHttpContent) {
  countBytes
}
```

Reply:
Yeah, good point. Will take your suggestion.